ts/WoltLabSuite/Core/Ui/Search.ts

```diff
@@ -0,0 +1,192 @@
+ * @copyright  2001-2019 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Page/Header/Fixed
+ */
+
+import * as EventHandler from "../Event/Handler";
+import * as UiAlignment from "./Alignment";
+import UiCloseOverlay, { Origin } from "./CloseOverlay";
+import UiDropdownSimple from "./Dropdown/Simple";
+import * as UiScreen from "./Screen";
+import * as Environment from "../Environment";
+import { identify } from "../Dom/Util";
+
+let _isMobile = false;
+let _scrollTop: number | undefined = undefined;
+
+const _isAcp = document.body.classList.contains("wcfAcp");
+const _pageHeader = document.getElementById("pageHeader")!;
+const _pageHeaderPanel = document.getElementById("pageHeaderPanel")!;
+const _pageHeaderSearch = document.getElementById("pageHeaderSearch")!;
+let _pageHeaderSearchMobile: HTMLElement | undefined = undefined;
+const _pageHeaderSearchInput = document.getElementById("pageHeaderSearchInput") as HTMLInputElement;
+const _topMenu = document.getElementById("topMenu")!;
+const _userPanelSearchButton = document.getElementById("userPanelSearchButton");
+
+/**
+ * Provides the collapsible search bar.
+ */
+function initSearchBar(): void {
+  _pageHeaderSearch.addEventListener("click", (event) => event.stopPropagation());
+
+  const searchType = document.querySelector(".pageHeaderSearchType") as HTMLElement;
+  const dropdownMenuId = identify(searchType);
+  const dropdownMenu = UiDropdownSimple.getDropdownMenu(dropdownMenuId)!;
+  dropdownMenu.addEventListener("click", (event) => {
+    // This prevents triggering the `UiCloseOverlay`.
+    event.stopPropagation();
+
+    UiDropdownSimple.close(dropdownMenuId);
+  });
+
+  _userPanelSearchButton?.addEventListener("click", (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (_pageHeader.classList.contains("searchBarOpen")) {
+      closeSearch();
+    } else {
+      openSearch();
+    }
+  });
+
+  UiCloseOverlay.add("WoltLabSuite/Core/Ui/Search", (origin, identifier) => {
+    if (origin === Origin.DropDown) {
+      const button = document.getElementById("pageHeaderSearchTypeSelect")!;
+      if (button.dataset.target === identifier) {
+        return;
+      }
+    }
+
+    closeSearch();
+
+    _pageHeaderSearchMobile?.setAttribute("aria-expanded", "false");
+  });
+}
+
+function initMobileSearch(): void {
+  const searchButton = document.getElementById("pageHeaderSearchMobile")!;
+  _pageHeaderSearchMobile = searchButton;
+
+  searchButton.addEventListener("click", (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (searchButton.getAttribute("aria-expanded") === "true") {
+      closeSearch();
+
+      searchButton.setAttribute("aria-expanded", "false");
+    } else {
+      if (Environment.platform() === "ios") {
+        _scrollTop = document.body.scrollTop;
+        UiScreen.scrollDisable();
+      }
```

Comment:
I assume this was copied from the old implementation and I believe this was because iOS tends to jump around the page. However this warrants a comment.